src/presets/ec-core/package.json

```diff
@@ -14,6 +14,9 @@
     "build": "rimraf build && run-p build:* -ln",
     "clean": "rimraf build dist",
     "dist": "npm run clean && cross-env NODE_ENV=production run-p build:* -ln",
+    "watch:scripts": "chokidar \"../../implementations/vanilla/**/*.js\" \"../**/*.js\" -i \"/node_modules|build|dist|config\\.js/\" -c \"npm run build:scripts\"",
+    "watch:styles": "chokidar \"../../implementations/vanilla/**/*.scss\" \"../../../../themes/**/*.scss\" \"../**/*.scss\" -i \"/node_modules|build|dist/\" -c \"npm run build:styles\"",
```

Comment:
nice to have these watchers, i wonder whether we could also have an "hot-reloading" in storybook, would be very handy

Reply:
Good remark. Actually it's there, but things are not as obvious and it's the subject I referred to today as "items to improve" in the dev task.

Two watch tasks are spawned in parallel [with lerna from project root](https://github.com/ec-europa/europa-component-library/blob/6e64dfcc13e79cd195779b8b34824cbe339912a1/package.json#L26-L28).

When any file of type .js or .css is updated in the vanilla implementation (presets watch task), ecl-builder rebuilds the corresponding styles and creates a new "build" version of the preset at `src/presets/(ec|eu)-core/build`. These build folders are indeed "sourced" through the watch task of the storybook instance https://github.com/ec-europa/europa-component-library/blob/6e64dfcc13e79cd195779b8b34824cbe339912a1/src/implementations/twig/storybook/package.json#L11 which is in its own turn also in watch mode (default to storybook actually)

The issues with watch tasks in presets in both ec/eu at the same time are several in my view:

1) log messages are lost because the way lerna spawns processes. i.e. we see `lerna [message]` types of logs, but it's not the same as running `yarn watch` in `src/presets/ec-core` for example (here in this comment) chokidar gives more details regarding its process. Error messages, warnings and useful feedback about rebuilding success on preset level is lost.

2) storybook watch task is not aware of the events emitted by chokidar which makes it so that ecl-builder successfully rebuilds the scss/javascript of the given preset, but it's not reflected in storybook until storybook application is reloaded fully and manually. If a file in storybook is touched, however, its active watcher in hot-reload will push changes to the storybook application successfully without any issues

For the first issue, a possible solution would be to update preset watch tasks to be like the storybook one: changing directory to the given preset folder and running the watch task without lerna.

For the second, I don't know yet. [chokidar](https://github.com/paulmillr/chokidar) has 'ready' event and we can potentially use it through a node api script. For the storybook watch task, I don't know: definitely welcome ideas how we can go lower than their utility script. Reading a bit, it goes to `node_modules/@storybook/html/dist/server/index.js` which is `https://github.com/storybookjs/storybook/blob/next/lib/core/server.js` and it seems on first look they use webpack. Not sure how we can listen to events emitted by their CLI (and even further emitting messages to the server )